src/RawRabbit/Operations/Publisher.cs

```diff
@@ -57,15 +56,19 @@ public Task PublishAsync<TMessage>(TMessage message, Guid globalMessageId, Publi
 					{
 						throw t.Exception;
 					}
+
+					channelTask.Result.BasicReturn += config.BasicReturn;
```

Comment:
Yes, that makes sense. I can see were having a `BasicReturn` handler would cause issues when it is not needed later on.